### DISCUSSION TITLE
Running mixed encryption mode

### DISCUSSION THREAD

**Discussion by chriseyhorn** [bluenviron/mediamtx#3335] [Questions]
### Question

Is it possible to run Webrtc with encryption both on and off?  We have a publisher using WHIP that doesn't support https but want all Whep client viewers to use https.

**Comment by aler9** [comment]
Hello, no, it's not possible and it won't be implemented due to a limited number of use cases. However you can accomplish this by disabling encryption in MediaMTX, and setupping a reverse proxy like Nginx or Caddy in front of MediaMTX. Users that want to use encryption can then connect to MediaMTX through the reverse proxy, while users that want to use plain connection can connect directly to MediaMTX.

[a comment by an automated account is not shown]
